src/Network/Wai/SAML2.hs

```diff
@@ -40,13 +40,15 @@ import Network.Wai.SAML2.Assertion
 import Network.Wai.SAML2.Error
 
 import System.IO.Unsafe (unsafePerformIO)
+import Data.ByteString (ByteString)
```

Comment:
Move this import up above the other `Data.` import. I'd also typically import this as qualified rather than with an import list (it makes no practical difference, but it would be more consistent in style with the rest of the code):
```suggestion
import qualified Data.ByteString as BS
```